test/unitTests/Tracking.js

```diff
@@ -0,0 +1,173 @@
+/* eslint-env mocha */
+/* eslint no-global-assign: 0 */
+var chai = require("chai");
+var expect = chai.expect;
+var sinon = require("sinon");
+var tracking = require("../../lib/events/tracking.js");
+var domUtils = require("../../lib/utils/dom.js");
+var assert = require("assert");
+
+
+
+describe("Unit Tests", function() {
+	var sandbox;
+	beforeEach(function () {
+		sandbox = sinon.sandbox.create();
+	});
+
+	afterEach(function () {
+		sandbox.restore();
+	});
+
+	describe("trackingHandler", function() {
+		it("should prevent default brrowser action", function() {
+
+			//arrange
+			var dataLayer = [];
+			window = sandbox.stub().returns("something");
+			window.setTimeout = sandbox.stub().returns("something");
+			window.dataLayer = dataLayer;
+			window.clearTimeout = sandbox.stub().returns("something");
+			var ev = {
+				target: {
+					nodeName: "A",
+					href: "www.google.com",
+					classList: "",
+					getAttribute: function(){
+						return this.href;
+					}
+				}
+			};
+			domUtils.find = sandbox.stub().returns(ev.target);
+
+
+			var spy = sandbox.spy(domUtils.enhanceEvent(ev), "preventDefault");
+
+			//act
+			tracking.handler(ev);
+
+			//assert
+			sandbox.assert.calledOnce(spy);
+		});
+	});
+
+	describe("sendTrackedEvent", function() {
+		it("will push to datalayer without callback", function() {
```

Comment:
Should we also have a test so that the callback is wrapped inside `dataLayer[0].eventCallback` and also one that tests that there's a fallback timeout?